ugvc/mrd/mrd_utils.py

```diff
@@ -772,3 +777,75 @@ def concat_dataframes(dataframes: list, outfile: str, n_jobs: int = 1):
         df = df.reset_index()
     df.to_parquet(outfile)
     return df
+
+
+def generate_synthetic_signatures(
+    signature_vcf: str, db_vcf: str, n_synthetic_signatures: int, output_dir: str
+) -> list[str]:
+    """
+    Generate synthetic signatures from a signature vcf file and a db vcf file
+
+    Parameters
+    ----------
+    signature_vcf: str
+        signature vcf file
+    db_vcf: str
+        db vcf file
+    n_synthetic_signatures: int
+        number of synthetic signatures to generate
+    output_dir: str
+        output directory
+
+    Returns
+    -------
+    synthetic_signatures: list[str]
+        list of synthetic control vcf files
+
+    """
+    # parse trinuc substitution distribution from signature vcf and db vcf
+    trinuc_signature = get_trinuc_substitution_dist(signature_vcf)
+    trinuc_db = get_trinuc_substitution_dist(db_vcf)
+
+    # allocate index and counter per synthetic signature
+    trinuc_dict = {}
+    for trinucsub in trinuc_signature.keys():
+        n_subs_signature = trinuc_signature[trinucsub]
+        n_subs_db = trinuc_db[trinucsub]
+        trinuc_dict[trinucsub] = {}
+        for i in range(n_synthetic_signatures):
+            f_index = f"index_to_sample_{i}"
+            f_counter = f"trinuc_counter_{i}"
+            trinuc_dict[trinucsub][f_index] = np.random.choice(range(0, n_subs_db), n_subs_signature, replace=False)
+            trinuc_dict[trinucsub][f_counter] = 0
+
+    # make output directory
+    os.makedirs(output_dir, exist_ok=True)
+
+    # read db vcf file and allocate output files and file handles
+    db_fh = pysam.VariantFile(db_vcf, "rb")
+    synthetic_signatures = []
+    file_handle_list = []
+    for i in range(n_synthetic_signatures):
+        output_vcf = pjoin(output_dir, f"syn{i}.vcf.gz")
```

Comment:
maybe include the basename of the db and of the signature?

Reply:
Done